src/main/scala/io/iohk/ethereum/network/p2p/messages/PV63.scala

```diff
@@ -25,6 +27,140 @@ object PV63 {
+      }
+
+      override def decode(rlp: RLPEncodeable): TransactionLog = rlp match {
+        case RLPList(loggerAddress, logTopics: RLPList, data) =>
+          TransactionLog(rlpDecode[ByteString](loggerAddress), logTopics.items.map(rlpDecode[ByteString]), rlpDecode[ByteString](data))
+        case _ => throw new RuntimeException("Cannot decode TransactionLog")
+      }
+    }
+  }
+
+  case class TransactionLog(loggerAddress: ByteString, logTopics: Seq[ByteString], data: ByteString) {
+    override def toString: String = {
+      s"""TransactionLog{
+         |loggerAddress: ${Hex.toHexString(loggerAddress.toArray[Byte])}
+         |logTopics: ${logTopics.map(e => Hex.toHexString(e.toArray[Byte]))}
+         |data: ${Hex.toHexString(data.toArray[Byte])}
+         |}
+       """.stripMargin
+    }
+  }
+
+  object Receipt {
+    implicit val rlpEndDec = new RLPEncoder[Receipt] with RLPDecoder[Receipt] {
+      override def encode(obj: Receipt): RLPEncodeable = {
+        import obj._
+        RLPList(rlpEncode[ByteString](postTransactionStateHash), cumulativeGasUsed,
+          rlpEncode[ByteString](logsBloomFilter), RLPList(logs.map(rlpEncode[TransactionLog]): _*))
+      }
+
+      override def decode(rlp: RLPEncodeable): Receipt = rlp match {
+        case RLPList(postTransactionStateHash, cumulativeGasUsed, logsBloomFilter, logs: RLPList) =>
+          Receipt(rlpDecode[ByteString](postTransactionStateHash), cumulativeGasUsed,
+            rlpDecode[ByteString](logsBloomFilter), logs.items.map(rlpDecode[TransactionLog]))
+        case _ => throw new RuntimeException("Cannot decode Receipt")
+      }
+    }
+  }
+
+  case class Receipt(
+    postTransactionStateHash: ByteString,
+    cumulativeGasUsed: BigInt,
+    logsBloomFilter: ByteString,
+    logs: Seq[TransactionLog]
+  ) {
+    override def toString: String = {
+      s"""
+         |Receipt{
+         |postTransactionStateHash: ${Hex.toHexString(postTransactionStateHash.toArray[Byte])}
+         |cumulativeGasUsed: $cumulativeGasUsed
+         |logsBloomFilter: ${Hex.toHexString(logsBloomFilter.toArray[Byte])}
+         |logs: $logs
+         |}
+       """.stripMargin
+    }
+  }
+
+  object Receipts {
+    implicit val rlpEndDec = new RLPEncoder[Receipts] with RLPDecoder[Receipts] {
+      override def encode(obj: Receipts): RLPEncodeable = {
+        import obj._
+        RLPList(receiptsForBlocks.map(r => RLPList(r.map{rlpEncode[Receipt]}: _*)): _*)
+      }
+
+      override def decode(rlp: RLPEncodeable): Receipts = rlp match {
+        case rlpList: RLPList => Receipts(rlpList.items.collect { case r: RLPList => r.items.map(rlpDecode[Receipt]) })
+        case _ => throw new RuntimeException("Cannot decode Receipts")
+      }
+    }
+
+    val code: Int = Message.SubProtocolOffset + 0x10
+  }
+
+  case class Receipts(receiptsForBlocks: Seq[Seq[Receipt]]) extends Message {
```

Comment:
A question about `Seq[Seq[...]]`. Does the outer sequence index determine which block the inner receipts are for?